samples/reading_events.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+from uuid import uuid4
+
+from esdbclient import (
+    EventStoreDBClient,
+    NewEvent,
+    StreamState,
+    exceptions,
+)
+
+client = EventStoreDBClient(uri="esdb://localhost:2113?tls=false")
+
+stream_name = str(uuid4())
+
+event_data = NewEvent(
+    type="some-event",
+    data=b'{"id": "1", "important_data": "some value"}',
+)
+
+# append 20 events
+for _ in range(20):
+    client.append_to_stream(
+        stream_name=stream_name,
+        current_version=StreamState.ANY,
+        events=event_data,
+    )
+
+# region read-from-stream
+events = client.get_stream(
+    stream_name=stream_name,
+    backwards=False,
+    stream_position=0,
+    limit=100,
+)
+
+# endregion read-from-stream
+
+# region iterate-stream
+for event in events:
+    # Doing something productive with the event
+    print(f"Event: {event}")
+# endregion iterate-stream
+
+
+# region read-from-stream-position
+events = client.get_stream(
+    stream_name=stream_name,
+    stream_position=10,
+    limit=20,
+)
+# endregion read-from-stream-position
+
+# region iterate-stream
+for event in events:
+    print(f"Event: {event}")
+# endregion iterate-stream
+
+
+try:
+    # region overriding-user-credentials
+    credentials = client.construct_call_credentials(
+        username="admin",
+        password="changeit",
+    )
+
+    stream = client.read_stream(
+        stream_name=stream_name,
+        stream_position=0,
+        credentials=credentials,
+    )
+    # endregion overriding-user-credentials
+except exceptions.GrpcError:
+    pass
+
+unknown_stream_name = str(uuid4())
+
+# region checking-for-stream-presence
+try:
+    events = client.get_stream(
+        stream_name=unknown_stream_name,
+        backwards=False,
+        stream_position=10,
+        limit=100,
+    )
+
+    for event in events:
+        print(event.data)
+except exceptions.NotFound:
+    pass
+# endregion checking-for-stream-presence
+
+# region reading-backwards
+events = client.get_stream(
+    stream_name=stream_name,
+    backwards=True,
+    limit=10,
+)
+
+for event in events:
+    print(f"Event: {event}")
+# endregion reading-backwards
+
+
+# region read-from-all-stream
```

Comment:
It might be a bit contentious, but actually in this client I have deemphasised the term "all stream" referring the "all" instead to "all recorded events in the database", and thereby reserving the term "stream" to the kind of thing you append events to and get events from, the things that have a given name, and a gapless sequence of integer positions, the stream positions.

I've heard for people who aren't already familiar with EventStoreDB that it's just a bit confusing.

I think that's because the "all stream" doesn't have stream positions, it has commit positions. You can't use any stream position to position yourself when reading the "all stream". Therefore, since the "all stream" doesn't have a gapless sequence of "stream positions", then either it isn't a stream, or the things that do have gapless sequences are something else. The same term is used for two different ideas, so it's objectively a confusion.

Anyway, let's see how it reads in the documentation?